### PR TITLE
ci: set up github action for black linting

### LINKED ISSUE #9: Implement Black Formatter
## Description
Add Black for formatting and linting, including a GitHub Workflow for checking the formatting

https://black.readthedocs.io/en/stable/integrations/github_actions.html

Can use the link above to get an github action workflow set up for all incoming PRs

### PR DESCRIPTION
**Pull request by jwu910**
Add github action for black to lint python code

Fixes #9

### CODE REVIEW

**Comment by jwu910** [comment]
@a-camarillo when you get a chance

File: pyproject.toml
Comment (a-camarillo) [inline pyproject.toml:1]: Is this required for pre-commit?

File: README.md
Comment (a-camarillo) [inline README.md:1]: does pre-commit call black? I'm guessing we shouldn't have to call both correct?

File: pyproject.toml
Comment (jwu910) [reply pyproject.toml:1]: It's a method to pass options into black from what I read

Comment (jwu910) [reply pyproject.toml:1]: Supports a bunch of other things to o

File: README.md
Comment (jwu910) [reply README.md:1]: Yeah pre commit has a bunch of hooks too, e we might not need the dependency for black but I'd have to check

**Review by a-camarillo — APPROVED** [review]

**Pull request merged**